### PR TITLE
cache VDF validation results (per process)

### PR DESCRIPTION
**Pull request by arvidn**
This isn't as effective as it may look since the complete block validation is done in a `ProcessPoolExecutor`, which means each process will have its own cache (presumably). We still check these multiple times in the main thread too though

### CODE REVIEW

File: chia/types/blockchain_format/vdf.py [hunk 1]
```diff
@@ -2,39 +2,46 @@
...
+from chia.types.blockchain_format.sized_bytes import bytes32, bytes100
 from chia.util.ints import uint8, uint64
 from chia.util.streamable import Streamable, streamable
 
 log = logging.getLogger(__name__)
 
-discriminant_cache: Dict[Tuple[bytes32, int], int] = {}
 
+@lru_cache(maxsize=20)
+def get_discriminant(challenge, size_bites) -> int:
```
Comment (richardkiss) [inline chia/types/blockchain_format/vdf.py:20]: `s/bites/bytes/` :)

**Review by richardkiss — APPROVED** [review]
Seems okay to me

File: chia/types/blockchain_format/vdf.py (hunk 1 above)
Comment (mariano54) [reply chia/types/blockchain_format/vdf.py:20]: Types for the arguments? :1st_place_medal:

Comment (mariano54) [reply chia/types/blockchain_format/vdf.py:20]: Maybe a bigger size too?

Comment (arvidn) [reply chia/types/blockchain_format/vdf.py:20]: heh, I think you wrote this code :)

**Pull request merged**